package.json

```diff
@@ -32,28 +32,32 @@
     "test:e2e": "jest --config ./test/jest-e2e.json"
   },
   "dependencies": {
-    "@nestjs/common": "^6.0.0",
-    "@nestjs/core": "^6.0.0",
-    "mqtt": "^3.0.0",
-    "reflect-metadata": "^0.1.12",
-    "rimraf": "^2.6.2",
-    "rxjs": "^6.3.3"
+    "mqtt": "4.3.1"
+  },
+  "peerDependencies": {
+    "@nestjs/common": "^6.10.11 || ^7.0.0 || ^8.0.0",
+    "@nestjs/core": "^6.10.11 || ^7.0.0 || ^8.0.0"
   },
   "devDependencies": {
-    "@nestjs/testing": "6.1.1",
+    "@nestjs/common": "^8.0.0",
+    "@nestjs/core": "^8.0.0",
+    "@nestjs/testing": "8.0.0",
     "@types/jest": "24.0.11",
     "@types/mqtt": "^2.5.0",
     "@types/node": "11.13.4",
     "@types/supertest": "2.0.7",
     "jest": "24.7.1",
     "prettier": "1.17.0",
+    "reflect-metadata": "^0.1.12",
+    "rimraf": "^2.6.2",
+    "rxjs": "^6.3.3",
```

Comment:
I could not install the deps (`npm i`) unless updating `rxjs` to the latest version (`^7.5.1`). Otherwise (without any deps modification) the following error is thrown (`node v17.0.1`, `npm 8.1.1`, `nest 8.1.6`):

```bash
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: nest-mqtt@0.1.7
npm ERR! Found: rxjs@6.6.7
npm ERR! node_modules/rxjs
npm ERR!   dev rxjs@"^6.3.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer rxjs@"^7.1.0" from @nestjs/common@8.2.4
npm ERR! node_modules/@nestjs/common
npm ERR!   dev @nestjs/common@"^8.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/ts/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ts/.npm/_logs/2022-01-05T16_27_20_121Z-debug.log
```

Reply:
This might be because I already use a newer major version in my app. And I also noticed that many of the other deps are not up to date.